### ISSUE TITLE
Matches multiple schemas when only one must validate

### ISSUE THREAD

**Issue by velara3** [BowlerHatLLC/vscode-as3mxml#358]
I've setup an HTML project and when I [created a launch debugger](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/Debug-a-transpiled-ActionScript-project-in-Visual-Studio-Code-with-Mozilla-Firefox) I'm getting the following message:

>Matches multiple schemas when only one must validate.

Here's the launch.json:

      {
         "version": "0.2.0",
         "configurations": [
            {
               "name": "Launch Firefox (localhost)",
               "type": "firefox",
               "request": "launch",
               "url": "http://localhost:8080/index.html",
               "webRoot": "${workspaceRoot}"
            }
         ]
      }

The warning is on the first bracket in the configurations array (line 4). 

Here is the asconfig.json:

      {
         "config": "js",
         "compilerOptions": {
            "source-map": true,
            "html-template": "template.html"
         },
         "files":
         [
            "src/Main.as"
         ]
      }

**Comment by joshtynjala** [comment]
I guess that the creators of the Firefox extension for VSCode must have changed something in their launch configuration format. I'll be happy to update the documentation with a better launch configuration, if you can figure out what's wrong and share the details here!

**Comment by joshtynjala** [comment]
Hmmm... I just tried copying your launch.json into one of my workspaces, and I'm not seeing the same error when I have the file open. It also launches correctly when I start the debugger.

**Comment by velara3** [comment]
I can confirm it's launching OK. It's not hitting the break points I've set in Main.as. I've restarted VSC and now the original message is gone:

>Matches multiple schemas when only one must validate.

FWIW I tried to use the Add Configuration button that VSC and that might have had something to do with the message but it's gone after restart. I'll keep working on the breakpoints issue if it is some how related. FYI The page is working.

**Comment by joshtynjala** [comment]
There is a known issue where breakpoints may not work when a Royale app is first starting up. It's like the Chrome/Firefox debugger in VSCode doesn't connect fast enough. However, breakpoints in event listeners that are triggered later (like a "click" listener) work fine. You may be running into that.

Details: https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/Common-issues#debugger-will-not-stop-at-breakpoints-on-startup-with-apache-royale

**Comment by velara3** [comment]
OK It's hitting breakpoints later on. I found a [note](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-firefox-debug#troubleshooting) about it in the Firefox debugger on it: 

>Breakpoints that should get hit immediately after the javascript file is loaded may not work the first time: You will have to click "Reload" in Firefox for the debugger to stop at such a breakpoint. This is a weakness of the Firefox debug protocol: VS Code can't tell Firefox about breakpoints in a file before the execution of that file starts.

**Comment by joshtynjala** [comment]
That's good info. I'll try to link to it as an official explanation.

**Comment by jkyeung** [comment]
> I've restarted VSC and now the original message is gone:
> 
> > Matches multiple schemas when only one must validate.

I'm just here to report that I encountered the same message while editing my `settings.json` file, and I couldn't figure out what the heck could be causing it. It was saying the problem was in the "launch" section of my settings, but I wasn't even editing that part of my settings. I thought I was going insane until I saw this comment from @velara3 and tried restarting VS Code. The message went away.

Moral of the story is: When you see weird warnings, restarting the editor is always worth a try.

**Comment by Alzarax** [comment]
> Moral of the story is: When you see weird warnings, restarting the editor is always worth a try.

>> Continually restarts VSC, warning persists. Oh well, It's a false positive anyway, for the most part.
Still, really does kick me in the stomach when I'm making fairly common edits to a .json file, however it can suck my toe.

**Comment by velara3** [comment]
There was a recent comment but it seems to have been removed. Since this post I've made a repository of working AS3 projects: 

https://github.com/velara3/as3

**Comment by armchairdeity** [comment]
I was having this issue, but when I restarted the editor it cleared up. However WRT to breakpoints not binding/firing, @velara3 - This was weird... SO MANY THINGS were 't working right.  However I had added the "Join By Process Name" config block for Node.JS. When I launched that debug profile I noticed that I had like 7 Node.js processes, so I went into Activity Monitor and killed them, and my breakpoints started binding again.

Weird stuff, y'all... but we figure it out one experience at a time. ;)
